### PR TITLE
Aaron Kitelinger Code Review

### PR DESCRIPTION
**Pull request by aaronkitelinger**
This is a program that takes in user's input on spinners and creates an activity for them using the user's options, of Activity type, Number of Participants, and Price range.
The code compiles and was available on time.
The commits I did on my own code are:

[https://github.com/wrytherUC/TaskRuler/commit/03afd5628c69cb500ea7a948d0f3350cfa77152e](url)
[https://github.com/wrytherUC/TaskRuler/commit/fc191dcd50bb9d0e55d7d25a9dcbd3c53293036b](url)
[https://github.com/wrytherUC/TaskRuler/commit/8b0b69ce52b7460891ea763e2457754f9637dd58](url)

I learned how to use the CircularProgessIndicator() to make the user know when the app is loading.
I learned how to user Spinners in multiple ways not only just seen in lecture videos.
I also learned the importance of implementing Strings from the res package as someone who doesn't speak a language wouldn't be able to use an app if they couldn't under the language.

### CODE REVIEW

File: app/src/main/java/com/ideastorm/v25001/IdeaStormApplication.kt [hunk 1]
```diff
@@ -12,6 +12,8 @@ class IdeaStormApplication : Application() {
     override fun onCreate() {
         super.onCreate()
 
+
+        // Koin Implementation
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/IdeaStormApplication.kt:16]: This comment isn't needed, as the code is readable.

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 2]
```diff
@@ -147,8 +147,12 @@ class MainActivity : ComponentActivity() {
      */
     @Composable
     fun ParticipantsSpinner() {
-        val participantOptions = listOf("One person", "Two person", "Group")
-        var participantText by remember { mutableStateOf("Number of participants") }
+        val participantOptions = listOf(
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:150]: Formatting can, and should, be performed automatically, and consistently, by the IDE.  Formatting code is really programmer's preference anyway, and doesn't necessarily decrease technical debt.

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 3]
```diff
@@ -196,18 +200,20 @@ class MainActivity : ComponentActivity() {
...
-            "Education",
-            "Recreational",
-            "Social",
-            "Diy",
-            "Charity",
-            "Cooking",
-            "Relaxation",
-            "Music",
-            "Busywork"
+            getString(R.string.Education),
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:205]: strings.xml is definitely a good idea... though in this case, I'd look for an online/dynamic source of data to populate this list.

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 4]
```diff
@@ -196,18 +200,20 @@ class MainActivity : ComponentActivity() {
...
+            getString(R.string.Diy),
+            getString(R.string.Charity),
+            getString(R.string.Cooking),
+            getString(R.string.Relaxation),
+            getString(R.string.Music),
+            getString(R.string.Busywork)
         )
-        var typeText by remember { mutableStateOf("Activity type") }
+
+        var typeText by remember { mutableStateOf(getString(R.string.ActivityType)) }
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:216]: String externalization is good.

BTW what is typeText?  Is there a more descriptive variable name to use here?

File: app/src/main/java/com/ideastorm/v25001/MainActivity.kt [hunk 5]
```diff
@@ -251,8 +257,12 @@ class MainActivity : ComponentActivity() {
      */
     @Composable
     fun PriceSpinner() {
-        val priceOptions = listOf("Free", "Low", "High")
-        var priceText by remember { mutableStateOf("Price range") }
+        val priceOptions = listOf(
+            getString(R.string.Free),
+            getString(R.string.Low),
+            getString(R.string.High)
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/MainActivity.kt:263]: See notes above.

File: app/src/main/java/com/ideastorm/v25001/service/ActivityService.kt [hunk 6]
```diff
@@ -13,6 +13,8 @@ interface IActivityService {
 }
 
 class ActivityService : IActivityService {
+
+    // function to return all Activities in the JSON
```
Comment (discospiff) [inline app/src/main/java/com/ideastorm/v25001/service/ActivityService.kt:17]: If putting comments over a function, use proper KDoc format.

**Comment by discospiff** [comment]
From this review, I'd harvest the string externalization.

No need to merge the entire branch, though.  Instead, let the IDE perform formatting for you, and use KDoc where appropriate.

**Comment by 3189080** [comment]
Aaron need some help on publishing

**Pull request closed without merging**